frontend/class-frontend.php

```diff
@@ -779,6 +779,10 @@ public function robots() {
 					$robots['index'] = 'noindex';
 				}
 			}
+			elseif ( is_404() ) {
```

Comment:
I want to place a side note on this solution. I'm not sure what the best solution for this is, perhaps it's something worth to think about:

`robots['follow']` is already set on [line 721](https://github.com/Yoast/wordpress-seo/blob/stories/DE/noindex-follow-404pages/frontend/class-frontend.php#L721), so it is not necessarily needed to set `$robots['follow']` here again.

As, according to the issue, `is_search()` and `is_404()` should display the same behavior, it might be best to merge the `elseif` with [line 738](https://github.com/Yoast/wordpress-seo/blob/stories/DE/noindex-follow-404pages/frontend/class-frontend.php#L738)

However, considering this is a big function of over 100 lines, the current solution might be the most safe and has the highest chance of being bug free.